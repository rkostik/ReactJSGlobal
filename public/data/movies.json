[
  {
    "title": "Pulp Fiction",
    "genre": "Action & Adventure",
    "year": "1994",
    "description": "Faced with life's cruel irony, the unpredictable stories of a well-dressed pair of low-level hitmen; a gangster's statuesque moll, and a double-crossing prizefighter become inextricably intertwined, as the small-time crooks, Honey Bunny and Pumpkin, summon up the courage to hold up their favourite L.A. diner.",
    "url": "images/pulp_fiction.png",
    "rating": "8.9",
    "runtime": "2h 34min"
  },
  {
    "title": "Bohemian Rhapsody",
    "genre": "Drama, Biography, Music",
    "year": "2018",
    "description": "Bohemian Rhapsody is a foot-stomping celebration of Queen, their music and their extraordinary lead singer Freddie Mercury. Freddie defied stereotypes and shattered convention to become one of the most beloved entertainers on the planet.",
    "url": "images/bohemian_rhapsody.png",
    "rating": "7.9",
    "runtime": "2h 14min"
  },
  {
    "title": "Kill Bill: Vol 2",
    "genre": "Oscar winning Movie",
    "year": "2004",
    "description": "Kill Bill: Volume 2 is a 2004 American neo Western martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who continues her campaign of revenge against the Deadly Viper Assassination Squad (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A.",
    "url": "images/kill_bill.png",
    "rating": "8.0",
    "runtime": "2h 17min"
  },
  {
    "title": "Avengers: War of Infinity",
    "genre": "Action & Adventure",
    "year": "2018",
    "description": "Avengers: Infinity War is a 2018 American superhero film based on the Marvel Comics superhero team the Avengers. ... In the film, the Avengers and the Guardians of the Galaxy attempt to prevent Thanos from collecting the six all-powerful Infinity Stones as part of his quest to kill half of all life in the universe.",
    "url": "images/avengers.png",
    "rating": "8.4",
    "runtime": "2h 29min"
  },
  {
    "title": "Inception",
    "genre": "Action & Adventure",
    "year": "2010",
    "description": "Inception is a 2010 science fiction action film written and directed by Christopher Nolan, who also produced the film with Emma Thomas, his wife. The film stars Leonardo DiCaprio as a professional thief who steals information by infiltrating the subconscious of his targets.",
    "url": "images/inception.png",
    "rating": "8.8",
    "runtime": "2h 28min"
  },
  {
    "title": "Reservoir Dogs",
    "genre": "Oscar winning Movie",
    "year": "1992",
    "description": "When a simple jewelry heist goes horribly wrong, the surviving criminals begin to suspect that one of them is a police informant. Six criminals, who are strangers to each other, are hired by a crime boss, Joe Cabot, to carry out a diamond robbery. ... They are completely sure that the robbery is going to be a success.",
    "url": "images/reservoir_dogs.png",
    "rating": "8.3",
    "runtime": "1h 39min"
  }
]