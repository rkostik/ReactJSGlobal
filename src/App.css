.App {
  text-align: center;
}

.App-header {
  background: url(/src/images/city2.webp);
  /*background-size: cover;
  min-height: 100vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;*/
  position:absolute;
  width: 1260px;
  height: 683px;
  left:28px;
  top:-48px;
  mix-blend-mode: normal;
  /*opacity: 0.3;
  filter: blur(8.15485);*/
}



.App-div{
  font-size: calc(10px + 2vmin);
  min-height: 300px;
  min-width: 300px;
}
