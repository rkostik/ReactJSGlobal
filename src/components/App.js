import React from 'react';
import Content from './Content/Content.jsx';
import styles from './App.module.css';
import Footer from './Footer/Footer.jsx';
import HeaderContent from './Header/HeaderContent.jsx';
import ModalMovieWrap from './Modal/ModalMovieWrap.jsx';
import ModalRemoval from './Modal/ModalRemoval.jsx';
import ModalCompletion from './Modal/ModalCompletion.jsx';

const App = () => (
    <>
        <div className={styles.main_wrap}>
            <div>
                <HeaderContent/>
                <Content/>
                <Footer/>
                <div className={styles.wrap}>
                    <ModalMovieWrap/>
                    <ModalRemoval/>
                    <ModalCompletion/>
                </div>
            </div>
        </div>
    </>
)

export default App