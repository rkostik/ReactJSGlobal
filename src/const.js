export const CONST_LIST =
    {
        SEARCH_PLACEHOLDER: 'What do you want to watch?',
        SEARCH_BTN_TXT: 'search',
        SEARCH_TITLE: 'find your movie',
        ERROR_MESSAGE: 'Something went wrong. We are trying to fix it.',
        BTN_ADD_TXT: '+add movie',
        PREFIX_LOGO: 'netflix',
        POSTFIX_LOGO: 'roulette',
        SEARCH_RESULT: ' movies found',
        CONFIRM_TITLE:'congratulations !',
        CONFIRM_TEXT:'The movie has been added to database successfully',
        REMOVAL_TITLE:'delete movie',
        REMOVAL_TEXT:'Are you sure you want to delete this movie?'
    }
;

export const CONST_ARR =
    {
        FILTER_LIST: ['all', 'documentary', 'comedy', 'horror', 'crime'],
        SORT_LIST: ['sort by', 'na']
    }

export const ADD_MOVIE_ARR={
    title: 'title',
    movie_url: 'movie url',
    url_default:'https://',
    rating_default:'7.8',
    runtime_default:'minutes',
    title_default:'movie name',
    genre: 'genre',
    release_date: 'release date',
    rating: 'rating',
    runtime: 'runtime',
    overview: 'overview',
    select_default:'Select genre',
    main_title: 'add movie',
    overview_description: 'Movie description',
    btn_reset: 'reset',
    btn_submit: 'submit',
    select_date: 'Select Date',
    main_title_edit: 'edit movie',
}
