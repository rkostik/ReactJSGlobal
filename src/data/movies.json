[
  {
    "title": "Pulp Fiction",
    "genre": "Action & Adventure",
    "year": "1994",
    "description": "",
    "url": "images/pulp_fiction.png"
  },
  {
    "title": "Bohemian Rhapsody",
    "genre": "Drama, Biography, Music",
    "year": "2018",
    "description": "",
    "url": "images/bohemian_rhapsody.png"
  },
  {
    "title": "Kill Bill: Vol 2",
    "genre": "Oscar winning Movie",
    "year": "2004",
    "description": "",
    "url": "images/kill_bill.png"
  },
  {
    "title": "Avengers: War of Infinity",
    "genre": "Action & Adventure",
    "year": "2018",
    "description": "",
    "url": "images/avengers.png"
  },
  {
    "title": "Inception",
    "genre": "Action & Adventure",
    "year": "2010",
    "description": "",
    "url": "images/inception.png"
  },
  {
    "title": "Reservoir Dogs",
    "genre": "Oscar winning Movie",
    "year": "1992",
    "description": "",
    "url": "images/reservoir_dogs.png"
  }
]