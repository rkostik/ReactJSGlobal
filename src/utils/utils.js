
export const timestampToString =((timestamp)=> Math.floor(timestamp / 60) + 'h ' + (timestamp - ((Math.floor(timestamp / 60)) * 60) +'min'));

export const genreConcat =((genres)=>genres.reduce((concat, current) => {
        return concat + ' & ' + current
    }));

export const getYear =((year)=> year.substr(0,4));


