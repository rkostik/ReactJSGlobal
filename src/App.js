import logo from './logo.svg';
import './App.css';
import Func_example from './components/Func_example/Func_example.jsx';
import Class_example from "./components/Class_example/Class_example";
import Button from "./components/Button/Button";
import React from "react";
import Text from "./components/Text/Text.jsx";


class App extends React.PureComponent {
    constructor(props) {
        super(props);
        this.handleBtnClick = this.handleBtnClick.bind(this);
        this.state = {
            isFunc: false,
            name: "Change",
            title: "Class component"
        };
    }

    handleBtnClick() {
        let isFunc = this.state.isFunc;
        let title = this.state.title;
        this.setState({isFunc: !isFunc, title:  isFunc? 'Func component' : 'Class component' })
    }

    render() {
        const isFunc = this.state.isFunc;
        let title = this.state.title;
        return (
            <div className="App-header">
                {isFunc && <Func_example title={title}/>}
                {!isFunc && <Class_example title={title}/>}
                <Button/>
                <Text name={this.state.name} onClick={this.handleBtnClick}/>
            </div>
        )
    }

}

export default App;
