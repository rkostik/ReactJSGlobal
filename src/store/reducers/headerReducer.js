import {initialState, initStateHeader} from '../../const.js';
import actions from '../actions/actionTypes.js';

const headerReducer = (state = {}, action) => {
    switch (action.type) {
        case actions.ADD_MOVIE_WIN:
            return {
                ...state,
                modalActive: action.modalActive
            }
        case actions.SEARCH_MOVIE:
            return {
                ...state,
                viewActive: action.viewActive
            }
        case actions.CLOSE_ADD_MOVIE:
            return {
                ...state,
                typeWindow: action.typeWindow,
                modalActive: action.modalActive,
                activeCalendar: action.activeCalendar
            }
        case actions.OPEN_CLICK:
            return {
                ...state,
                activeCalendar: !state.activeCalendar
            }
        case actions.DDM_CLOSE_CLICK:
            return {
                ...state,
                //deleteActive: action.deleteActive,
                ddmActive: action.ddmActive
            }
        case actions.DDM_OPEN_CLICK:
            return {
                ...state,
                item: action.item,
                ddmActive: !action.ddmActive
            }
        case actions.MOVIE_CLICK:
            return {
                ...state,
                item: action.item,
                viewActive: action.viewActive
            }
        case actions.EDIT_CLICK:
            return {
                ...state,
                editActive: action.editActive,
                typeWindow: action.typeWindow,
                ddmActive: action.ddmActive,
                calendarDate: action.calendarDate,
                movie:action.movie
            }
        case actions.DEL_WINDOW_CLICK:
            return {
                ...state,
                item: action.item,
                deleteActive: action.deleteActive
            }
        case actions.DELETE_CLICK:
            return {
                ...state,
                deleteActive: action.deleteActive,
                ddmActive: action.ddmActive
            }
        case actions.DEL_WINDOW_CLICK:
            return {
                ...state,
                deleteActive: action.deleteActive
            }
        case actions.POPULATE_MOVIE_LIST:
            return {
                ...state,
                movies: action.movies,
                amount: action.amount,
                sortValue: action.sortValue
            }
        case actions.FILTER_MOVIES:
            return {
                ...state,
                movies: action.movies,
                amount: action.amount,
                filterName: action.filterName
            }
        case actions.POPULATE_MOVIE_LIST_ERR:
            return {
                ...state,
                errMessage: action.errMessage
            }
        case actions.CHANGE_DATE:
            return {
                ...state,
                calendarDate: action.calendarDate,
                activeCalendar: action.activeCalendar
            }
        case actions.SELECT_GENRE:
            return {
                ...state,
                optionSelected: action.optionSelected
            }
        case actions.RESET_FORM:
            return {
                ...state,
                calendarDate: action.calendarDate,
                optionSelected: action.optionSelected
            }
        case actions.AFTER_SUBMIT:
            return {
                ...state,
                calendarDate: action.calendarDate,
                optionSelected: action.optionSelected,
                completeActive: action.completeActive
            }
        case actions.CLOSE_CONFIRM_WIN:
            return {
                ...state,
                completeActive: action.completeActive,
                deleteActive: action.deleteActive
            }
        case actions.SET_SEARCH_TEXT:
            return {
                ...state,
                searchText: action.searchText
            }
        default:
            return state
    }
}

export default headerReducer