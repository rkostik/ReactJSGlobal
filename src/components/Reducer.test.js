import {shallow} from "enzyme";
import React from 'react';
import App from './App';
import data from '../jest/mockData.js';
import {
    addMovieClick,
    closeAddMovieClick,
    ddmCloseClick,
    editMovieList
} from '../store/actions/actionCreators';
import headerReducer from '../store/reducers/headerReducer.js';
import {initialState} from '../const.js';
import {stringToDate} from '../utils/utils.js';


it('render without crashing',()=>{
    shallow(<App/>);
});

describe('test reducer', () => {
    it('returns the same state on an unhandled action', () => {
        expect(headerReducer(initialState, {type: '_NULL'})).toMatchSnapshot();
    });

    it('handles ADD_MOVIE_WIN action', () => {
        expect(headerReducer(initialState, addMovieClick())).toEqual({
            ...initialState,
            modalActive: true
        });

    });

    it('handles CLOSE_ADD_MOVIE action', () => {

        expect(headerReducer(initialState, closeAddMovieClick())).toEqual({
            ...initialState,
            modalActive: false,
            typeWindow: 'add',
            activeCalendar: false
        });

    });

    it('handles DDM_CLOSE_CLICK action', () => {

        expect(headerReducer(initialState, ddmCloseClick())).toEqual({
            ...initialState,
            ddmActive: false
        });

    });

    it('handles EDIT_CLICK action', () => {

        expect(headerReducer(initialState, editMovieList(data))).toEqual({
            ...initialState,
            editActive: true,
            typeWindow: 'edit',
            ddmActive: false,
            calendarDate: stringToDate(data.release_date),
            movie: data
        });

    });

})
