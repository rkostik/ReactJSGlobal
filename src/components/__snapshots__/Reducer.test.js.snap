// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test reducer returns the same state on an unhandled action 1`] = `
Object {
  "activeCalendar": false,
  "amount": 6,
  "calendarDate": null,
  "completeActive": false,
  "const_list": Object {
    "BTN_ADD_TXT": "+add movie",
    "CONFIRM_TEXT": "The movie has been added to database successfully",
    "CONFIRM_TITLE": "congratulations !",
    "ERROR_MESSAGE": "Something went wrong. We are trying to fix it.",
    "POSTFIX_LOGO": "roulette",
    "PREFIX_LOGO": "netflix",
    "REMOVAL_TEXT": "Are you sure you want to delete this movie?",
    "REMOVAL_TITLE": "delete movie",
    "SEARCH_BTN_TXT": "search",
    "SEARCH_PLACEHOLDER": "What do you want to watch?",
    "SEARCH_RESULT": " movies found",
    "SEARCH_TITLE": "find your movie",
  },
  "ddmActive": false,
  "deleteActive": false,
  "editActive": false,
  "errMessage": "",
  "filterName": "all",
  "isSelected": true,
  "item": -1,
  "modalActive": false,
  "movie": Array [],
  "movies": Array [
    Object {
      "budget": 55000000,
      "genres": Array [
        "Drama",
        "Romance",
      ],
      "id": 337167,
      "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
      "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
      "release_date": "2018-02-07",
      "revenue": 136906000,
      "runtime": 106,
      "tagline": "Don't miss the climax",
      "title": "Fifty Shades Freed",
      "vote_average": 6.1,
      "vote_count": 1195,
    },
    Object {
      "budget": 150000000,
      "genres": Array [
        "Mystery",
        "Science Fiction",
        "Thriller",
      ],
      "id": 335984,
      "overview": "Thirty years after the events of the first film, a new blade runner, LAPD Officer K, unearths a long-buried secret that has the potential to plunge what's left of society into chaos. K's discovery leads him on a quest to find Rick Deckard, a former LAPD blade runner who has been missing for 30 years.",
      "poster_path": "https://image.tmdb.org/t/p/w500/gajva2L0rPYkEWjzgFlBXCAVBE5.jpg",
      "release_date": "2017-10-04",
      "revenue": 259239658,
      "runtime": 163,
      "tagline": "There's still a page left.",
      "title": "Blade Runner 2049",
      "vote_average": 7.3,
      "vote_count": 3955,
    },
    Object {
      "budget": 200000000,
      "genres": Array [
        "Fantasy",
        "Adventure",
        "Science Fiction",
      ],
      "id": 181808,
      "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
      "poster_path": "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
      "release_date": "2017-12-13",
      "revenue": 1325937250,
      "runtime": 152,
      "tagline": "The Saga Continues",
      "title": "Star Wars: The Last Jedi",
      "vote_average": 7.1,
      "vote_count": 4732,
    },
    Object {
      "budget": 200000000,
      "genres": Array [
        "Action",
        "Adventure",
        "Fantasy",
        "Science Fiction",
      ],
      "id": 284054,
      "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan \\"special forces\\"), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "poster_path": "https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date": "2018-02-13",
      "revenue": 1245257672,
      "runtime": 134,
      "tagline": "Long live the king",
      "title": "Black Panther",
      "vote_average": 7.3,
      "vote_count": 3788,
    },
    Object {
      "budget": 175000000,
      "genres": Array [
        "Adventure",
        "Comedy",
        "Family",
        "Animation",
      ],
      "id": 354912,
      "overview": "Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel's family history.",
      "poster_path": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
      "release_date": "2017-10-27",
      "revenue": 700920729,
      "runtime": 105,
      "tagline": "The celebration of a lifetime",
      "title": "Coco",
      "vote_average": 7.8,
      "vote_count": 3619,
    },
    Object {
      "budget": 175000000,
      "genres": Array [
        "Adventure",
        "Science Fiction",
        "Action",
      ],
      "id": 333339,
      "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
      "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
      "release_date": "2018-03-28",
      "revenue": 0,
      "runtime": 140,
      "tagline": "A better reality awaits.",
      "title": "Ready Player One",
      "vote_average": 8.1,
      "vote_count": 617,
    },
  ],
  "optionSelected": "",
  "searchText": "",
  "sortValue": "release date",
  "typeWindow": "add",
  "viewActive": true,
}
`;
